modules/seqexec/server/src/main/scala/seqexec/server/SeqexecEngine.scala

```diff
@@ -786,11 +761,28 @@ object SeqexecEngine extends SeqexecConfiguration {
 
     val smartGcal = smartGcalEnable.fold(initSmartGCal(smartGCalHost, smartGCalDir), IO.unit)
 
-    smartGcal *>
-      epicsInit *>
-      (for {
-        now <- IO(LocalDate.now)
-      } yield Settings(site,
+    def gpiClient: cats.effect.Resource[IO, GpiClient[IO]] =
+      if (gpiControl === FullControl) {
+        GpiClient.gpiClient[IO](gpiUrl, ec)
+      } else {
+        GpiClient.simulatedGpiClient(ec)
+      }
+
+    def ghostClient: cats.effect.Resource[IO, GhostClient[IO]] =
+      if (gpiControl === FullControl) {
+        GhostClient.ghostClient[IO](ghostUrl, ec)
+      } else {
+        GhostClient.simulatedGhostClient(ec)
+      }
+
+    def settings: IO[Settings] = {
+      val giapiClients = for {
+        now   <- cats.effect.Resource.liftF(IO(LocalDate.now))
+        gpi   <- gpiClient
+        ghost <- ghostClient
+      } yield (now, gpi, ghost)
+      giapiClients.use { case (now, gpi, ghost) =>
```

Comment:
After some testing I think thi is bogus. The resource will be released after settings is created. Any ideas how to properly do this?

Reply:
`Resource[IO, Settings]`?